packages/simcore-sdk/src/simcore_sdk/node_ports_common/filemanager.py

```diff
@@ -243,6 +248,20 @@ class UploadedFolder:
     ...
 
 
+async def _generate_checksum(
+    path_to_upload: Path | UploadableFileObject, is_directory: bool
+) -> SHA256Str | None:
+    checksum: SHA256Str | None = None
+    if is_directory:
+        return checksum
+    if isinstance(path_to_upload, Path):
+        async with aiofiles.open(path_to_upload, mode="rb") as f:
+            checksum = SHA256Str(await create_sha256_checksum(f))

```

Comment:
There is an endpoint on the API server which uploads files through the api. That endpoint is a bottleneck because the api server actually holds the file in memory. I added endpoints to do multipart uploads. Hence, I mainly added this for backwards compatibility.